### PR TITLE
[tech] add api status

### PR DESCRIPTION
**Pull request by Moomoo-Seukla**
Add version number
The status API allows you to see:

- the application's configuration
- date of the last message received by the SNCF
- date of the last attempt to send to RabbitMQ
- date of the last successful write to the RabbitMQ queue
All dates is in `map[string]time.Time`, I preferred to use a map for date score in case we need to add another one. 

I have added :
 - scoreboard
 -  handler 
 - and some code moves in main.go

Ticket : [ND-1469](https://jira.kisio.org/browse/ND-1469)

NB: For the moment, not tests are written.

### CODE REVIEW

File: internal/webstomp/webstomp.go [hunk 1]
```diff
@@ -65,7 +65,7 @@ func (c *Client) connect() {
 	}
 }
 
-func (c *Client) Consume(consumer messageConsumer) {
+func (c *Client) Consume(consumer messageConsumer, scoreboard *ScoreBoard) {
```
Comment (pbougue) [inline internal/webstomp/webstomp.go:68]: Same remark than the amqp Client: this shouldn't be part of the interface, it's an internal mechanism.
In that case, there is no other choice but to pass it to the Client constructor, as this is looping over messages (no func "ConsumeOneMessage")

File: internal/webstomp/webstomp.go [hunk 2]
```diff
@@ -86,6 +86,7 @@ func (c *Client) Consume(consumer messageConsumer) {
 				c.logger.Infof("failed to aknowledge message, err: %v\n", err)
```
Comment (pbougue) [inline internal/webstomp/webstomp.go:86]: It was there before:
```suggestion
				c.logger.Infof("failed to acknowledge message, err: %v\n", err)
```

File: internal/webstomp/scoreboard.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package webstomp
```
Comment (pbougue) [inline internal/webstomp/scoreboard.go:1]: Why is that related to webstomp specifically ?
It's related to the status, so I'd let it right in `internal`.

The same goes for `status_handler` and `webserver`.

File: internal/webstomp/status_handler.go [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+	if websocketconn, err := Dial(stompOpts.Target, stompOpts.Protocol); err == nil {
+		r.IsWebStompConnActive = true
+		websocketconn.Close()
+	}
+}
+
+func (r *responseStatus) setStatusScoreBoard(mapScore map[string]time.Time) {
+	r.LastMessageReceiveSNCF = mapScore["lastMessageReceiveSNCF"]
+	r.LastAttempSendRabbitMQ = mapScore["lastAttempSendRabbitMQ"]
+	r.LastSuccessfulWriteRabbitMQ = mapScore["lastSuccessfulWriteRabbitMQ"]
```
[the remark this answers is not in the document]
Comment (Moomoo-Seukla) [reply internal/webstomp/status_handler.go:63]: `mapScore[key]` return `nil` if the key not found.

[the remark this answers is not in the document]
Comment (pbougue) [reply internal/webstomp/status_handler.go:63]: OK, but will the /status be OK (the field there with a "nil" value) ?

[the remark this answers is not in the document]
Comment (Moomoo-Seukla) [reply internal/webstomp/status_handler.go:63]: if the field is `nil` the display is `0001-01-01T00:00:00Z`, I test it

[the remark this answers is not in the document]
Comment (datanel) [reply internal/webstomp/status_handler.go:63]: > `mapScore[key]` return `nil` if the key not found.

`mapScore[key]` does not return nil but the "zero" of type `time.Time`

https://play.golang.org/p/d2nbXvuA63w

File: internal/datecontainer/datecontainer.go [hunk 5]
```diff
@@ -0,0 +1,37 @@
+package datecontainer
+
```
Comment (pbougue) [inline internal/datecontainer/datecontainer.go:2]: Not sure what's more idiomatic: Should you name the file after the structure it contains (when it's dedicated)?
So `safe_date_container.go` here?

**Review by pbougue — APPROVED** [review]

File: internal/datecontainer/datecontainer.go (hunk 5 above)
Comment (Moomoo-Seukla) [reply internal/datecontainer/datecontainer.go]: Done

**Review by pbougue — APPROVED** [review]
